Flow.Launcher.Core/Updater.cs

```diff
@@ -29,89 +29,69 @@ public Updater(string gitHubRepository)
+                var currentVersion = Version.Parse(Constant.Version);
 
-            Log.Info($"|Updater.UpdateApp|Future Release <{newUpdateInfo.FutureReleaseEntry.Formatted()}>");
+                Log.Info($"|Updater.UpdateApp|Future Release <{newUpdateInfo.FutureReleaseEntry.Formatted()}>");
 
-            if (newReleaseVersion <= currentVersion)
-            {
-                if (!silentUpdate)
-                    MessageBox.Show("You already have the latest Flow Launcher version");
-                updateManager.Dispose();
-                return;
-            }
+                if (newReleaseVersion <= currentVersion)
+                {
+                    if (!silentUpdate)
+                        MessageBox.Show("You already have the latest Flow Launcher version");
+                    updateManager.Dispose();
+                    return;
+                }
 
-            if (!silentUpdate)
-                api.ShowMsg("Update found", "Updating...");
+                if (!silentUpdate)
+                    api.ShowMsg("Update found", "Updating...");
 
-            try
-            {
                 await updateManager.DownloadReleases(newUpdateInfo.ReleasesToApply);
+
+                await updateManager.ApplyReleases(newUpdateInfo);
+
+                if (DataLocation.PortableDataLocationInUse())
+                {
+                    var targetDestination = updateManager.RootAppDirectory + $"\\app-{newReleaseVersion.ToString()}\\{DataLocation.PortableFolderName}";
+                    FilesFolders.CopyAll(DataLocation.PortableDataPath, targetDestination);
+                    if (!FilesFolders.VerifyBothFolderFilesEqual(DataLocation.PortableDataPath, targetDestination))
+                        MessageBox.Show("Flow Launcher was not able to move your user profile data to the new update version. Please manually " +
+                            $"move your profile data folder from {DataLocation.PortableDataPath} to {targetDestination}");
+                }
+                else
+                {
+                    await updateManager.CreateUninstallerRegistryEntry();
+                }
+
+                var newVersionTips = NewVersinoTips(newReleaseVersion.ToString());
+
+                Log.Info($"|Updater.UpdateApp|Update success:{newVersionTips}");
+
+                if (MessageBox.Show(newVersionTips, "New Update", MessageBoxButton.YesNo) == MessageBoxResult.Yes)
+                {
+                    UpdateManager.RestartApp(Constant.ApplicationFileName);
+                }
             }
             catch (Exception e) when (e is HttpRequestException || e is WebException || e is SocketException)
             {
                 Log.Exception($"|Updater.UpdateApp|Check your connection and proxy settings to github-cloud.s3.amazonaws.com.", e);
-                updateManager.Dispose();
+                api.ShowMsg("Update Fail!", "Check your connection and proxy settings to github-cloud.s3.amazonaws.com.");
```

Comment:
just wording while we are at it, can we update title to "Updated failed" and subtitle to "Check your connection and try updating proxy settings to github-cloud.s3.amazonaws.com."

Reply:
Sure